lib/writeResources.js

```diff
@@ -45,64 +41,52 @@ function writeResources(gltf, options) {
     options.separateShaders = defaultValue(options.separateShaders, false);
     options.dataUris = defaultValue(options.dataUris, false);
 
-    var promises = [];
-
     ForEach.image(gltf, function(image, i) {
-        promises.push(writeImage(gltf, image, i, options));
+        writeImage(gltf, image, i, options);
         ForEach.compressedImage(image, function(compressedImage) {
-            promises.push(writeImage(gltf, compressedImage, i, options));
+            writeImage(gltf, compressedImage, i, options);
         });
     });
 
     ForEach.shader(gltf, function(shader, i) {
-        promises.push(writeShader(gltf, shader, i, options));
+        writeShader(gltf, shader, i, options);
     });
 
     // Buffers need to be written last because images and shaders may write to new buffers
-    return Promise.all(promises)
-        .then(function() {
-            removeUnusedElements(gltf);
-            mergeBuffers(gltf);
-            return writeBuffer(gltf, gltf.buffers[0], 0, options);
-        })
-        .then(function() {
-            return gltf;
-        });
+    removeUnusedElements(gltf);
+    mergeBuffers(gltf);
+    writeBuffer(gltf, gltf.buffers[0], 0, options);
+    return gltf;
 }
 
 function writeBuffer(gltf, buffer, i, options) {
     if (defined(options.bufferStorage)) {
         options.bufferStorage.buffer = buffer.extras._pipeline.source;
-        return Promise.resolve();
```

Comment:
Do we still want to early return here?

Reply:
Ah yes, thanks for catching that. Fixed.